app/models/agents/website_agent.rb

```diff
@@ -166,6 +169,25 @@ def validate_options
       validate_web_request_options!
     end
 
+    def validate_consider_http_success_option!
+      consider_success = options["consider_http_error_success"]
+      if consider_success != nil
```

Comment:
I'd recommend

```ruby
if consider_success.present?
```

Also, in the error messages, could you indicate the key that is causing the problem?  For example, "If used, consider_http_error_success must be an Array" or "consider_http_error_success contains a duplicate http status code".

Finally, I'd recommend allowing strings or integers in the array.  We can make sure the code handles either.